apps/io-wallet-user-func/src/infra/attestation-service/android/attestation.ts

```diff
@@ -8,6 +8,19 @@ import { X509Certificate, createPublicKey } from "crypto";
 import * as jose from "jose";
 import * as pkijs from "pkijs";
 
+import { AndroidDeviceDetails } from ".";
+
+class AndroidAttestationError extends Error {
+  name = "AndroidAttestationError";
+  constructor(message: string, deviceDetails?: AndroidDeviceDetails) {
+    super(
+      deviceDetails
+        ? `[Android Attestation Error] ${message} - Device details: ${JSON.stringify(deviceDetails)}`
+        : `[Android Attestation Error] ${message}`,
+    );
+  }
+}
+
 /**
```

Comment:
I think we should map the error in `toHttpError` function in `/src/infra/http/error.ts` otherwise it will return a generic 500 error